tools/version_check.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+# Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+# For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+# All errors are manually handled
+set -uo pipefail
+
+L_GIT_DIR=$(git rev-parse --show-toplevel)
+if [ ! -d "$L_GIT_DIR" ]; then L_GIT_DIR=$(readlink -f "$(dirname "$0")"/..); fi
+
+. "$L_GIT_DIR"/tools/locate_bins.sh
+require_commands head sed cut
+
+# first argument is the "binname --version" output
+# second argument is the minimum acceptable version
+minimum_version () {
+  VERSION=$(echo "$1" | $call_sed -e "s/[^0-9]*\([0-9]\+\)\.\([0-9]\+\)\.\([0-9]\+\).*/\\1.\\2.\\3/")
```

Comment:
you are aware that not all commands are following sem ver? So in fact this function is not generic it says it's.
Also - this function is so overcomplicated...
you can just:
1. read the sem ver to array ( see how `IFS='.' read -r -a <<< $string` would work) 
```
[pholat@pholat-mudita ~]$ echo $string
1.2.3
[pholat@pholat-mudita ~]$ IFS='.' read -r -a array <<< "$string"
[pholat@pholat-mudita ~]$ echo ${array[1]}
2
```
2. check against the second array

With variable checks enabled it will be done without any external commands and in something like 3 lines.

Reply:
I believe the utilities that need to have their versions checked do follow semantic versioning.

Your version checking solution is quite nice but it relies on non-standard command behavior for `read`.  Meanwhile `awk` is standardized, it's not installed by default on many Linux systems, and it's behavior is inconsistent across implementations due to it's complexity.

I agree my solution is not very "clean" but it is highly portable.

Reply:
I've remove the version checking modification from this PR.